src/coreclr/src/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -4066,9 +4066,18 @@ internal static Type GetTypeFromCLSIDImpl(Guid clsid, string? server, bool throw
                 }
             }
 
-            // For target invocation exceptions, the exception is wrapped.
-            flags |= BindingFlags.DoNotWrapExceptions;
-            object? ret = InvokeMember(memberName, flags, null, target, aArgs, aParamMod, null, null);
+            object? ret;
+            try
+            {
+                // The IDispatch codepath doesn't support BindingFlags.DoNotWrapExceptions.
```

Comment:
`InvokeMember` is a public API. This sounds like a bug that we should fix. 

Note that `DoNotWrapExceptions` is a new flag in .NET Core and we forgot to add the proper handling for it on this path.

Reply:
>Note that DoNotWrapExceptions is a new flag in .NET Core and we forgot to add the proper handling for it on this path.

Didn't realize that. This makes altering the code path more reasonable.